src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9251,6 +9248,9 @@ private ExpressionSyntax ParseTerm(Precedence precedence)
                 case SyntaxKind.NewKeyword:
                     expr = this.ParseNewExpression();
                     break;
+                case SyntaxKind.StackAllocKeyword:
+                    expr = this.ParseStackAllocExpression();
+                    break;
```

Comment:
The level is same as `new . . . `
Yes, this would be indexing of stackalloc expression. The grammar allows for precisely
`stackalloc <type> [ <expr> ] `    that did not change.

The precedence will need to be documented, unless it is already documented. We really just relaxing where the expr could be used. And not by much right now.

Reply:
And since it is relaxing, it is not a breaking change. So I think this is the correct thing to do.

---
In reply to: [133259285](https://github.com/dotnet/roslyn/pull/21006#discussion_r133259285) [](ancestors = 133259285)